src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs

```diff
@@ -43,33 +43,41 @@ public class JenkinsArtifact : AgentService, IArtifactExtension
             executionContext.Output(StringUtil.Loc("RMJenkinsBuildId", jenkinsDetails.BuildId));
 
             Stream downloadedStream = null;
-            using (HttpClient client = new HttpClient())
+            using (HttpClientHandler handler = new HttpClientHandler())
             {
-                SetupHttpClient(client, jenkinsDetails.AccountName, jenkinsDetails.AccountPassword);
-                var downloadArtifactsUrl =
-                    new Uri(
-                        string.Format(
-                            CultureInfo.InvariantCulture,
-                            "{0}/job/{1}/{2}/artifact/{3}/*zip*/",
-                            jenkinsDetails.Url,
-                            jenkinsDetails.JobName,
-                            jenkinsDetails.BuildId,
-                            jenkinsDetails.RelativePath));
-
-                executionContext.Output(StringUtil.Loc("RMPrepareToGetFromJenkinsServer"));
-                HttpResponseMessage response = client.GetAsync(downloadArtifactsUrl).Result;
-
-                if (response.IsSuccessStatusCode)
+                using (HttpClient client = new HttpClient(handler))
                 {
-                    downloadedStream = response.Content.ReadAsStreamAsync().Result;
-                }
-                else if (response.StatusCode == HttpStatusCode.NotFound)
-                {
-                    throw new ArtifactDownloadException(StringUtil.Loc("RMNoArtifactsFound", jenkinsDetails.RelativePath));
-                }
-                else
-                {
-                    throw new ArtifactDownloadException(StringUtil.Loc("RMDownloadArtifactUnexpectedError"));
+                    handler.ServerCertificateCustomValidationCallback = (message, certificate, chain, sslPolicyErrors) =>
```

Comment:
move this outside using (HttpClient) block as it is initializing the handler only